### DISCUSSION TITLE
Can we run it against colab kernel?

### DISCUSSION THREAD

**Discussion by Mahesha999** [microsoft/vscode-jupyter#9753] [Questions and Answers] [answered]
I wanted to replicate exact versions of packages listed in one of the project's Pipfile inside google colab.  Since google colab does not allow to do so, I tried to somewhat convoluted way. I started vscode server inside colab and accessed it using ngrok using ColabCode. The code is in [this](https://colab.research.google.com/drive/1Yu_ctumurM5ApvVp882GfWst5G3RoSLI) notebook. With this code running I was able to run vscode in browser.

Now from in-browser vscode's terminal:

   1. Install `pipenv` with `pip install pipenv`.

   2. Change directory inside google drive. Say `cd /content/drive/MyDrive/NewFolder` (Given `NewFolder` exists at that path).
 
   3. Next, `git clone` the desired github repo (with `Pipfile`) inside google drive `NewFolder`.

   4. Start virtual environment with `pipenv shell`

   5. Install python packages from `Pipfile` with `pipenv install`. Assuming this also contained `jupyter`

   6. Choose the newly created python environment inside vscode
   
   7. Now opening and running juyter notebook is giving following error: 
   
![image](https://user-images.githubusercontent.com/4706946/164787757-67a5ee57-658a-4eb1-ad5f-039f925f46f7.png)

(If this works, it will allow me have vscode's debugging capability, jupyter's visualization capability, colab's GPUs all running against replicable environment captured by Pipfile)
 
**Update**

If I change the python interpreter of notebook to default one instead of pipenv (at the top right corner of notebook), the notebook works. So it seems to fail to work only for pipenv.

**Comment by rchiodo** [comment, marked as answer]
Looks to me like the jupyter config on the VM is requesting google specific stuff. 

This message here:

```
The `kernel_class` trait instance must be a type, but 'google.colab._kernel.kenrel' could not be imported.
```

That's likely because the config for the jupyter server on the box specifies kernels are based on google colab kernels. However you're starting your kernel without any google stuff in the path. 

You might try deleting the jupyter_notebook_config.py on the machine (if that's findable). It's the thing that's likely specifying the base class for a kernel.
